proto/v1/structure/state.proto

```diff
@@ -0,0 +1,881 @@
+  }
+
+  // Structure of the rules of the image click input.
+  message RuleSpec {
+
+    // Different rules which can be applied on the image click input.
+    oneof rule_type {
+
+      // A rule to check the region availability.
+      IsInRegionSpec is_in_region = 1;
+    }
+
+    // Structure of a single is in region.
+    message IsInRegionSpec {
+      string input_region = 1;
+    }
+  }
+}
+
+// Answer type: RatioExpression.
+// Structure of the ratio input interaction.
+message RatioExpressionInputInstance {
+
+  // A set of arguments to customize the ratio expression input.
+  CustomizationArgs customization_args = 1;
+
+  // A list of answer groups of the ratio expression input interaction instance.
+  repeated AnswerGroup answer_groups = 2;
+
+  // The default outcome of the ratio expression input.
+  Outcome default_outcome = 3;
+
+  // A list of hints for the ratio expression input.
+  repeated Hint hints = 4;
+
+  // The solution of the ratio expression input.
+  Solution solution = 5;
+
+  // Structure of the customization args of the ratio expression input.
+  message CustomizationArgs {
+
+    // A placeholder for the ratio expression input.
+    SubtitledHtml placeholder = 1;
+
+    // The number of terms in the ration expression.
+    int32 number_of_terms = 2;
+  }
+
+  // Structure of the solution of the ratio expression input.
+  message Solution {
+
+    // A set of common items for the solution in all interactions.
+    BaseSolution base_solution = 1;
+
+    // The correct answer of this ratio expression input.
+    RatioExpression correct_answer = 2;
+  }
+
+  // Structure of a single answer group of ratio expression input.
+  message AnswerGroup {
+
+    // A set of common items for the answer groups in all the interactions.
+    BaseAnswerGroup base_answer_group = 1;
+
+    // A list of all the rules of the ratio expression input.
+    repeated RuleSpec rule_specs = 2;
+  }
+
+  // Structure of the rules of the ratio expression input.
+  message RuleSpec {
+
+    // Different rules which can be applied on the ratio expression input.
+    oneof rule_type {
+
+      // A rule to check the equality.
+      EqualsSpec equals = 1;
+
+      // A rule to check the equivalent.
+      IsEquivalentSpec is_equivalent = 2;
+
+      // A rule to check the number of terms.
+      HasNumberOfTermsEqualToSpec has_number_of_terms_equal_to = 3;
+    }
+
+    // Structure of the equal rule check.
+    message EqualsSpec {
+      RatioExpression input = 1;
+    }
+
+    // Structure of the equivalent rule check.
+    message IsEquivalentSpec {
+      RatioExpression input = 1;
+    }
+
+    // Structure of the numer of terms equal rule check.
+    message HasNumberOfTermsEqualToSpec {
+      uint32 input_term_count = 1;
+    }
+  }
+}
+
+// Answer type: none (N/A since no answers are submitted for this interaction).
+// Structure of a single end exploration type interaction.
+message EndExplorationInstance {
```

Comment:
Don't we need `SubtitledHtml` for end exploration as there is some content shown in the demo exploration?

```
  End Exploration:
    classifier_model_id: null
    content:
      content_id: content
      html: <p>Congratulations, you have finished!</p>
```